pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java

```diff
@@ -163,6 +169,11 @@ public CompletableFuture<List<String>> getChildrenFromStore(String path) {
                             future.completeExceptionally(ex);
                             return null;
                         });
+                    } else if (code == Code.CONNECTIONLOSS) {
+                        // There is the chance that we caused a connection reset by sending or requesting a batch
+                        // that passed the max ZK limit. Retry with the individual operations
+                        executor.schedule(() -> getChildrenFromStore(path),
```

Comment:
We need pass the result of `getChildrenFromStore` to `future` in this.

Reply:
Good point, addressed.